src/frontend/src/session.rs

```diff
@@ -581,6 +582,15 @@ pub struct SessionImpl {
 
     /// Last idle instant
     last_idle_instant: Arc<Mutex<Option<Instant>>>,
+
+    /// Whether in a Read-Write Transaction. We need this because we don't really start a
+    /// transaction when the transaction is read-write but cursors should only be declared
+    /// within transaction blocks. This flag is used to determine whether the session is in
+    /// a transaction
+    pub in_rw_txn: Mutex<bool>,
```

Comment:
Why not reusing `txn`? This looks duplicated

Reply:
Currently, due to the workaround for Read-Write transaction, the `txn`'s state is set to `State::Initial`(i.e., no transaction in progress) thus the session is not aware that it's in a transaction, when there's a RW txn running. To make the session aware that it's in a RW txn, we can either add a flag like this PR, or introduce a new state to https://github.com/risingwavelabs/risingwave/blob/921e59e15d4cd9b8dd33722c3176eac4342f3ce3/src/frontend/src/session/transaction.rs#L73 to imply that there's a transaction running. I think the previous implementation seems simpler?

Reply:
@BugenZhao Any suggestions? 🙏

Reply:
As the read-write transaction is not actually supported in our system, is it still necessary to check whether there's a running transaction when declaring the cursor? Will there be any trouble if we tolerate this?